woocommerce/woocommerce-gutenberg-products-block
# [Store Customization] Update patterns to use the AI selected verticals for images

## What

Update all patterns to ensure images are populated with Verticals (an API built to help designers and developers choose vertical-specific images for use in templates and automatically generated content). The displayed images should dynamically change depending on what the user decides to add as a description for their business on their AI settings screen and also during the upcoming onboarding screen.

Fixes woocommerce/woocommerce-blocks#10623

## Why

This is required to ensure all of our patterns have AI-managed images in place.



## Testing Instructions



_Please consider any edge cases this change may have, and also other areas of the product this may impact._

1. Create a new [JN install](https://jurassic.ninja/) with both Jetpack and Woocommerce installed and activated.
3. Install and activate WooCommerce Blocks from this zip file:  [woocommerce-gutenberg-products-block.zip](https://github.com/woocommerce/woocommerce-blocks/files/12611550/woocommerce-gutenberg-products-block.zip)

5. In your wp-admin, you should see the following section for Jetpack:

<img width="1803" alt="Screenshot 2023-09-11 at 18 55 38" src="https://github.com/woocommerce/woocommerce-blocks/assets/15730971/fc804759-f724-4fa2-829d-7bfd441962ec">

6. Click on "Set up Jetpack" and make sure the connection is successful:

<img width="1994" alt="Screenshot 2023-09-11 at 18 58 30" src="https://github.com/woocommerce/woocommerce-blocks/assets/15730971/661ee4c5-a0ac-40fd-9e5e-835a0e4c1cb7">

7. From your admin screen, click on Tools > Plugin editor: 

<img width="337" alt="Screenshot 2023-09-11 at 18 34 23" src="https://github.com/woocommerce/woocommerce-blocks/assets/15730971/c353bd13-bf00-4bc8-93d6-d827ca99997f">

8. Select the WooCommerce Blocks plugin and add the following within your woocommerce-blocks/woocommerce-gutenberg-products-block.php file and save:

```php
function connect_to_endpoint() {
	update_option( 'woocommerce_blocks_allow_ai_connection', true );

	$request = new WP_REST_Request( 'POST', '/wc/store/patterns' );

	$request->set_param( 'business_description', 'A store that sells organic food' );

	$response = rest_do_request( $request )->get_data();

	error_log( print_r( $response, true ) );
}

add_action( 'admin_init', 'connect_to_endpoint' );
```

9. Create a new post
10. Insert the Featured Category Triple pattern (or any other you want with at least one image).
11. Make sure the images on the pattern match the business description you provided: 'A store that sells organic food'
12. Go back to the plugin editor and update the business description over there to something else (e.g. books, clothes, sports gear etc.) and save
13. Return to your post editor, save it, and refresh the page.
14. Insert the same Featured Category Triple pattern again
15. Notice how all images were updated to now match what you selected as the new description for your store*

*ps: there's a limitation in terms of the number of images available in the Verticals API to populate these patterns, so not all store descriptions will necessarily return the expected images just yet.


## Screenshots or screencast




https://github.com/woocommerce/woocommerce-blocks/assets/15730971/5f3ae0fa-483b-4c57-8833-73252372abbc



## WooCommerce Visibility


Required:


## Checklist

Required:

Conditional:

## Changelog


N/A
